File: nVideo/Controllers/HomeController.cs
using System.Collections.Generic;
using System.Diagnostics;
using Microsoft.AspNetCore.Mvc;
using Interfaces;
using Models;
using ViewModels;

#nullable enable

namespace nVideo.Controllers
{
    public class HomeController : Controller
    {
        private readonly IAllCatalog _catalog;

        private IEnumerable<Catalog_Entity> GetThridRandom => 
            _catalog.GetRandomItem();

        public HomeController(IAllCatalog catalog) =>
            _catalog = catalog;

        public IActionResult Index(){
            var model = new HomeViewModel();

            model.ForCarousel = _catalog.GetCarouselItems();
            model.ForFeaturedBlock = _catalog.GetFeaturedItems();
            model.ForNewProductsBlock = _catalog.GetNewItems();
            model.ForThumbnailBlock = GetThridRandom;

            return View(model);
        }      

        public IActionResult Privacy(){
            return View();
        }

        public IActionResult City(string city)
        {
            HttpContext.Response.Cookies.Append("City", city);

            return RedirectToAction("Index");
        }

        [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
        public IActionResult Error(){
            return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
        }
    }
}

File: fVideo/nVideo/Controllers/HomeController.fs
namespace nVideo.Controllers

open System
open System.Diagnostics
open Microsoft.AspNetCore.Mvc
open Models
open nVideo.Models
open System.Linq
open Microsoft.EntityFrameworkCore
open Application
open DataLayer

type HomeController (context : AppDbContext) =
    inherit Controller()

    let products = context.Products.Include(fun x -> x.Image)

    member this.Index () =
        DbProduct.Initial context 
        let carousel = products.OrderByDescending(fun x -> x.Raiting).Take(10).ToList()
        let featured = products.OrderByDescending(fun x -> x.Raiting).Take(10).ToList()
        let newProducts = products.OrderByDescending(fun x -> x.Date).Take(10).ToList()
        let thumbnail = products.AsParallel().Take(3).ToList()

        
        let model = {
            Carousel = carousel
            Featured = featured
            Thumbnail = thumbnail
            NewProducts = newProducts
        }
        this.View(model)
    member this.Privacy () =
        this.View()

    

    [<ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)>]
    member this.Error () =
        let reqId = 
            if isNull Activity.Current then
                this.HttpContext.TraceIdentifier
            else
                Activity.Current.Id

        this.View({ RequestId = reqId })
